bin/pygac-run

```diff
@@ -90,19 +74,62 @@ def str2scanline(string):
 def validate_args(args):
     if args.end_line > 0 and args.start_line > args.end_line:
         raise ValueError('Start Scanline > End Scanline')
+    if args.config and not os.path.isfile(args.config):
+        raise FileNotFoundError(
+            'The provided config file "%s" does not exist!'
+            % args.config)
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(
         description='Read, calibrate and navigate NOAA AVHRR GAC data')
-    parser.add_argument('filename', type=str, help='GAC file to be processed')
+    parser.add_argument('path', type=str, help='Path to GAC file(s) to be processed '
+                        '(Can be a file, directory or tar-archive)')
     parser.add_argument('start_line', type=str2scanline,
                         help='First scanline to be processed (0-based)')
     parser.add_argument('end_line', type=str2scanline,
                         help='Last scanline to be processed (0-based, '
                              'set to 0 for the last available scanline)')
+    parser.add_argument('-c', '--config', type=str,
+                        help='pygac config file to be used')
     args = parser.parse_args()
     validate_args(args)
-
-    reader = check_file_version(args.filename)
-    reader(args.filename, args.start_line, args.end_line)
+    if args.config:
+        os.environ['PYGAC_CONFIG_FILE'] = args.config
+    from pygac.l1c_factory import L1cFactory
+    l1c_factory = L1cFactory()
+    path = args.path
+    start_line = args.start_line
+    end_line = args.end_line
+    if os.path.isfile(path): 
+        if tarfile.is_tarfile(path):
+            # archive
+            logger.info('Open archive "%s"' % str(path))
+            with tarfile.open(path) as a:
```

Comment:
Good idea, I would also introduce a decorator for the `l1c_factory` (or `l1c_processor`) call which implements the try except logging clause to avoid the code repetition.